src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -460,7 +460,7 @@ private static IEnumerable<PathSegment> GetPathSegments(string httpRequestUri, I
 
         public virtual Parameter BuildConstructorParameter(InputParameter operationParameter)
         {
-            var parameter = BuildParameter(operationParameter);
+            var parameter = BuildParameter(operationParameter, operationParameter is { Kind: InputOperationParameterKind.Client, Type: InputEnumType enumType } ? _typeFactory.CreateType(enumType.EnumValueType).FrameworkType : null);
```

Comment:
We need a TODO or comment here.

Reply:
This change can pass the `cadl-ranch` `multi-client` case, but will create breaking change in autorest project `Multiplefiles`. I'm still trying to make a better work-around.

Reply:
reverted this change, we choose to keep enum type except for `api-version`